### PR TITLE
Janusgraph visualization

### PR DESCRIPTION
**Pull request by fuss86**

### CODE REVIEW

File: src/main/java/org/carlspring/strongbox/janusgraph/graph/schema/StrongboxSchema.java [hunk 1]
```diff
@@ -23,9 +24,10 @@
     private static final Logger logger = LoggerFactory.getLogger(StrongboxSchema.class);
 
     @Inject
-    public void createSchema(JanusGraphFactory.Builder jgf)
+    public void createSchema(JanusGraph jg)
     {
-        JanusGraph jg = jgf.open();
+        //drop(jg);
```
Comment (fuss86) [inline src/main/java/org/carlspring/strongbox/janusgraph/graph/schema/StrongboxSchema.java]: temporary solution if the schema already exists

File: src/main/java/org/carlspring/strongbox/janusgraph/graph/schema/StrongboxSchema.java [hunk 2]
```diff
@@ -66,4 +70,67 @@ private void applySchemaChanges(JanusGraphManagement jgm)
...
+        }
+        catch (BackendException e)
+        {
+            throw new RuntimeException(e);
+        }
+        jgm.commit();
+    }
+
+    // TODO FIXME
+    private void sampleExample(JanusGraph jg)
```
Comment (fuss86) [inline src/main/java/org/carlspring/strongbox/janusgraph/graph/schema/StrongboxSchema.java]: just for POC of visualization

File: src/main/java/org/carlspring/strongbox/janusgraph/graph/JanusGraphConfig.java [hunk 3]
```diff
@@ -19,14 +16,17 @@
 {
 
     @Bean
-    public Builder janusGraphFactory(CassandraDaemon cassandra,
-                                     CassandraEmbeddedProperties cassandraEmbeddedProperties)
+    JanusGraph janusGraph(GremlinServer gremlinServer)
```
Comment (sbespalov) [inline src/main/java/org/carlspring/strongbox/janusgraph/graph/JanusGraphConfig.java]: @fuss86 the first thing is that we shouldn't expose `JanusGraph` as a `@Bean`, it should be like it was before.
In terms of JPA `JanusGraph` is more like `EntityManager` and `Builder` is like `EntityManagerFactory`

And the second thing is that this is probably fine to use `GremlinServer` for graph explorer but we shouldn't use it within the application itself.

File: src/main/java/org/carlspring/strongbox/janusgraph/graph/JanusGraphConfig.java [hunk 4]
```diff
@@ -19,14 +16,17 @@
...
+    {
+        return (JanusGraph) gremlinServer.getServerGremlinExecutor()
+                                         .getGraphManager()
+                                         .getGraph("graph");
+    }
+
+    @Bean
+    GraphTraversalSource g(JanusGraph janusGraph)
```
Comment (sbespalov) [inline src/main/java/org/carlspring/strongbox/janusgraph/graph/JanusGraphConfig.java]: this shouldn't be a `@Bean` as well.

File: src/main/java/org/carlspring/strongbox/janusgraph/graph/gremlin/GremlinConfig.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.Resource;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@Configuration
+@ComponentScan
+public class GremlinConfig
```
Comment (sbespalov) [inline src/main/java/org/carlspring/strongbox/janusgraph/graph/gremlin/GremlinConfig.java]: this configuration should be optional, like we currently have it in strongbox. So you can disable graph explorer and `GremlinServer` which provide remote protocol to get access to DB instance.

File: src/main/java/org/carlspring/strongbox/janusgraph/graph/gremlin/GremlinConfig.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+{
+
+    @Bean(destroyMethod = "stop")
+    @ConditionalOnMissingBean
+    GremlinServer gremlinServer(
+            CassandraDaemon cassandra,
+            @Value("classpath:conf/gremlin-server/gremlin-server.yaml") Resource gremlinServerConf,
+            @Value("classpath:conf/gremlin-server/janusgraph-cql-es-server.properties") Resource janusGraphBackendProperties)
```
Comment (sbespalov) [inline src/main/java/org/carlspring/strongbox/janusgraph/graph/gremlin/GremlinConfig.java]: we better to expose configuration settings that we need with `application,yaml`, you can have a look at original `JanusGraphConfig` as an example.

File: src/main/java/org/carlspring/strongbox/janusgraph/graph/schema/StrongboxSchema.java [hunk 7]
```diff
@@ -23,9 +24,10 @@
     private static final Logger logger = LoggerFactory.getLogger(StrongboxSchema.class);
 
     @Inject
-    public void createSchema(JanusGraphFactory.Builder jgf)
+    public void createSchema(JanusGraph jg)
```
Comment (sbespalov) [inline src/main/java/org/carlspring/strongbox/janusgraph/graph/schema/StrongboxSchema.java]: @fuss86 can you please leave it like it was originally? (the reason is in above comments)

File: src/main/resources/conf/gremlin-server/janusgraph-cql-es-server.properties [hunk 8]
```diff
@@ -0,0 +1,160 @@
+# Copyright 2019 JanusGraph Authors
```
Comment (sbespalov) [inline src/main/resources/conf/gremlin-server/janusgraph-cql-es-server.properties]: we don't need this configuration (see above)

File: src/main/resources/scripts/empty-sample.groovy [hunk 9]
```diff
@@ -0,0 +1,30 @@
+// Copyright 2019 JanusGraph Authors
```
Comment (sbespalov) [inline src/main/resources/scripts/empty-sample.groovy]: this probably should be under `src/main/groovy`

File: src/main/java/org/carlspring/strongbox/janusgraph/graph/JanusGraphConfig.java (hunk 3 above)
Comment (sbespalov) [reply src/main/java/org/carlspring/strongbox/janusgraph/graph/JanusGraphConfig.java]: ok, regarding the "first" part please ignore it. I've looked again and probably `JanusGraph` instance assumed to be thread safe and can be the `EntityManagerFactory` :)
I was just confused with that `JanusGraph` interface extends `Transaction` interface, but now I see that the implementation is `ThreadLocal` based and seems thread safe.

The "second"  still valid, so we shouldn't make `JanusGraph` depends on `GremlinServer`.

File: src/main/java/org/carlspring/strongbox/janusgraph/graph/JanusGraphConfig.java (hunk 4 above)
Comment (sbespalov) [reply src/main/java/org/carlspring/strongbox/janusgraph/graph/JanusGraphConfig.java]: ignore this as well

File: src/main/java/org/carlspring/strongbox/janusgraph/graph/schema/StrongboxSchema.java (hunk 7 above)
Comment (sbespalov) [reply src/main/java/org/carlspring/strongbox/janusgraph/graph/schema/StrongboxSchema.java]: ignore

File: src/main/java/org/carlspring/strongbox/janusgraph/graph/JanusGraphConfig.java (hunk 3 above)
Comment (fuss86) [reply src/main/java/org/carlspring/strongbox/janusgraph/graph/JanusGraphConfig.java]: done

**Comment by fuss86** [comment]
It resolves

```
[ ] Management consoles (@fuss86)
  [x]  Investigate whether there is a management console for Janusgraph
```

from https://github.com/strongbox/strongbox/issues/1573

**Pull request merged**